spread/client.go

```diff
@@ -84,38 +84,47 @@ func (c *Client) dialOnReboot(prevUptime time.Time) error {
 	waitConfig.Timeout = 5 * time.Second
 	uptimeChanged := 3 * time.Second
 
+	// close old session
+	c.sshc.Close()
 	for {
-		before := time.Now()
+		// Try to connect to the rebooting system, note that
+		// waitConfig is not well honored by golang, it is
+		// set to 5sec above but in reality it takes ~60sec
+		// before the code times out.
 		sshc, err := ssh.Dial("tcp", c.addr, &waitConfig)
-		if err != nil {
-			// It's gone.
-			continue
-		}
-
-		c.sshc.Close()
-		c.sshc = sshc
-		currUptime, err := c.getUptime()
-		if err != nil {
-			// Not available uptime yet
-			continue
+		if err == nil {
+			// once successfully connected, check uptime to
+			// see if the reboot actually happend
+			c.sshc = sshc
```

Comment:
Don't we need to close the old session here instead of out of the loop, similar to how the old code was doing it? Otherwise everything we pass by this logic in the loop we'll keep the previous connection open and dangling.